Make stripping configurable, correct CC/CXX/CFLAGS/CXXFLAGS handling, add
missing include

diff --git a/Makefile b/Makefile
index 1433167..38bb7cc 100644
--- a/Makefile
+++ b/Makefile
@@ -28,12 +28,13 @@ ifeq ($(MAKECMDGOALS),Makedep.force)
 MAKE_DIST=000
 endif

+GCC_STRIP ?= -s
 BASH=bash
 PERL_BADLANG=x
 export PERL_BADLANG

 CXXX=$(CXD_assert)
-LDALL=$(LDXX) -s $(LDFLAGS) $(LIBS)
+LDALL=$(LDXX) $(GCC_STRIP) $(LDFLAGS) $(LIBS)
 ifeq ($(ENABLE_DEBUG), no)
 CXXX=$(CXD_no)
 endif
@@ -46,8 +47,8 @@ CXXX=$(CXD_checker)
 LDALL=checkerg++ $(LDFLAGS) $(LIBS)
 endif

-CXXALL=$(CXXX) $(CXXFLAGS) $(CXXFLAGSB)
-CCALL =$(CXXX) $(CXXFLAGS) $(CXXFLAGSB)
+CXXALL=$(CXX) $(CXXFLAGS) $(CXXFLAGSB)
+CCALL =$(CC) $(CFLAGS) $(CXXFLAGSB)
 CXDFAL=$(CXXFLAGS) $(CXXFLAGSB) $(LDFLAGS) $(LIBS)

 .PHONY: clean dist dist-noautoconf dist-more allclean distclean dist-install
@@ -67,7 +68,7 @@ ifeq ($(MAKE_DIST),)
   #
   # echo to config.h is needed to make config.h younger than config.h.in;
   # configure doesn't change the timestamp of config.h if it's unchanged.
-  Makedep config.h: configure Makehelp.in config.h.in ccdep.pl sam2p_version.h; ./configure MAKE=$(MAKE) --enable-gif --enable-lzw && echo "" >>config.h
+  Makedep config.h: configure Makehelp.in config.h.in ccdep.pl sam2p_version.h;
   Makehelp:;
   include Makehelp  # Defines CXX. File generated by ./configure.
   include Makedep   # Contains ALL+=... .
@@ -101,8 +102,8 @@ IDE_MODES := release debug
 IDE_TARGETS_debug := $(patsubst %,%.yes, $(TARGETS))
 IDE_TARGETS_release := $(TARGETS)

-CXD_assert =$(CXX) -s -O2
-CXD_no     =$(CXX) -s -DNDEBUG -O3
+CXD_assert =$(CXX) $(GCC_STRIP) $(CXXFLAGS)
+CXD_no     =$(CXX) $(GCC_STRIP) $(CXXFLAGS)
 # CXD_yes    =$(CXX) $(GFLAG) -lefence
 CXD_yes    =$(CXX) $(GFLAG)
 CXD_checker=checkerg++ $(GFLAG)
